internal/core/src/common/Vector.h

```diff
@@ -100,28 +112,81 @@ class ColumnVector final : public BaseVector {
     }
 
     void*
-    GetRawData() {
+    RawValueAt(size_t index, size_t size_of_element) override {
+        return reinterpret_cast<char*>(GetRawData()) + index * size_of_element;
+    }
+
+    bool
+    ValidAt(size_t index) override {
+        return valid_values_[index];
+    }
+
+    void*
+    GetRawData() const {
         return values_->Data();
     }
 
     void*
     GetValidRawData() {
-        return valid_values_->Data();
+        return valid_values_.data();
     }
 
     template <typename As>
-    const As*
+    As*
     RawAsValues() const {
-        return reinterpret_cast<const As*>(values_->Data());
+        return reinterpret_cast<As*>(values_->Data());
+    }
+
+    bool
+    IsBitmap() const {
+        return is_bitmap_;
     }
 
  private:
+    bool is_bitmap_;
```

Comment:
this is not a good implment, can just using BitmapVector as a sub class of baseVector

Reply:
Yes, but I think this PR is large enough. Replace `ColumnVector` with `BitmapVector`  in other Exprs would introduce a large number of code modifications. It's better to implement `BitmapVector` in a new PR.

Reply:
See #37075